test/test-users.js

```diff
@@ -0,0 +1,402 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+const spies = require('chai-spies');
+const mongoose = require('mongoose');
+const app = require('../server').app;
+
+const {
+  User
+} = require('../users/models');
+const should = chai.should();
+
+var UrlPattern = require('url-pattern');
+const pattern = {
+  "/users": new UrlPattern('/users'),
+  "/users/:userId": new UrlPattern('/users/:userId')
+}
+
+chai.use(chaiHttp);
+chai.use(spies);
+
+describe('Test /user endpoints with empty database', () => {
+  beforeEach((done) => {
+    // Approach #1 Drop database and `ensureIndexes()`
+    mongoose.connection.db.dropDatabase()
```

Comment:
Into with having an `ensureIndexes` method on `User`. Maybe call it `ensureTestIndexes` or something like that just so it's clear method is for testing.

Would like to retain the before, beforeEach, afterEach, after steps mentioned above:

```
before(function() {
    return runServer(TEST_DATABASE_URL);
  });

  beforeEach(function() {
    return promiseThatSeedsDb();
  });

  afterEach(function() {
    return tearDownDb();
  });

  after(function() {
    return closeServer();
  })

```

So the db gets seeded before each test (`ensureIndex`) and tear it down after each test.